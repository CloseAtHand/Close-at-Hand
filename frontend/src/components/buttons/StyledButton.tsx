import { Pressable, StyleSheet, Text, View } from "react-native";
import { COLORS, FONTSIZE } from "../../shared/styles/STYLES";

const colorMatch = {
	SkyBlue: COLORS.Black,
	Turquoise: COLORS.Black,
	CarrotRed: COLORS.White,
	Black: COLORS.White,
	White: COLORS.Black,
};

interface Props {
	title: string;
	onPress(): void;
	backgroundColor?: keyof typeof colorMatch;
}

const StyledButton: React.FC<Props> = ({ title, onPress, backgroundColor }) => {
	return (
		<View style={styles.container}>
			<Pressable
				style={({ pressed }) => [
					styles.button,
					pressed && styles.pressed,
					{ backgroundColor: COLORS[backgroundColor ?? "SkyBlue"] },
					backgroundColor === "White" && styles.whiteBackground,
				]}
				onPress={onPress}
				testID="button"
			>
				<Text
					style={[
						styles.text,
						{ color: colorMatch[backgroundColor ?? "SkyBlue"] },
					]}
				>
					{title}
				</Text>
			</Pressable>
		</View>
	);
};

export default StyledButton;

const styles = StyleSheet.create({
	container: {
		alignItems: "center",
	},
	button: {
		borderColor: COLORS.White,
		borderRadius: 5,
		borderWidth: 1,
		minHeight: 45,
		margin: 10,
		padding: 10,
	},
	pressed: {
		opacity: 0.75,
	},
	text: {
		fontSize: FONTSIZE.ExtraSmall,
		textAlign: "center",
		fontWeight: "bold",
		width: "auto",
	},
	whiteBackground: {
		borderWidth: 1,
		borderRadius: 5,
		borderColor: COLORS.Gray,
	},
});
