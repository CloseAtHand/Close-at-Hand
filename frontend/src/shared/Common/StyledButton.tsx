import { Pressable, StyleSheet, Text, View } from "react-native";
import COLORS from "../constant/COLORS";

const colorMatch = {
	SkyBlue: COLORS.Black,
	CarrotRed: COLORS.White,
	Black: COLORS.White,
	White: COLORS.Black,
};

interface Props {
	title: string;
	onPress: () => void;
	backgroundColor?: keyof typeof colorMatch;
}

const StyledButton: React.FC<Props> = ({ title, onPress, backgroundColor }) => {
	return (
		<View style={styles.container}>
			<Pressable
				style={({ pressed }) => [
					styles.button,
					pressed && styles.pressed,
					{ backgroundColor: COLORS[backgroundColor ?? "SkyBlue"] },
				]}
				onPress={onPress}
				testID="button"
			>
				<Text
					style={[
						styles.text,
						{ color: colorMatch[backgroundColor ?? "SkyBlue"] },
					]}
				>
					{title}
				</Text>
			</Pressable>
		</View>
	);
};

export default StyledButton;

const styles = StyleSheet.create({
	container: {
		alignItems: "center",
	},
	button: {
		borderRadius: 5,
		borderWidth: 1,
		minHeight: 45,
		margin: 10,
		padding: 10,
	},
	pressed: {
		opacity: 0.75,
	},
	text: {
		fontSize: 16,
		textAlign: "center",
		fontWeight: "bold",
		width: "auto",
	},
});
