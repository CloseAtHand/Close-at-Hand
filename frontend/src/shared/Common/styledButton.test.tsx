import "@testing-library/jest-native/";
import { render, fireEvent } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import StyledButton from "./StyledButton";

describe("StyledButton", () => {
	const INITIAL_VALUE = "value";

	it("스냅샷 테스트", () => {
		const tree = renderer
			.create(<StyledButton title="Click Me" onPress={jest.fn()} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("버튼에 텍스트가 올바르게 렌더링됩니다.", () => {
		const { getByText } = render(
			<StyledButton title={INITIAL_VALUE} onPress={jest.fn()} />,
		);

		expect(getByText(INITIAL_VALUE)).toBeTruthy();
	});

	it("onPress가 버튼을 터치할 때마다 호출됩니다.", () => {
		const mockPressFn = jest.fn();
		const { getByTestId } = render(
			<StyledButton title={INITIAL_VALUE} onPress={mockPressFn} />,
		);

		const button = getByTestId("button");

		fireEvent.press(button);
		expect(mockPressFn).toHaveBeenCalledTimes(1);

		fireEvent.press(button);
		expect(mockPressFn).toHaveBeenCalledTimes(2);
	});
});
