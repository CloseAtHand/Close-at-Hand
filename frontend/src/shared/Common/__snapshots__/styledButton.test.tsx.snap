// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton 스냅샷 테스트 1`] = `
<View
  style={
    {
      "alignItems": "center",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#8DE0F8",
        "borderColor": "#ffffff",
        "borderRadius": 5,
        "borderWidth": 1,
        "margin": 10,
        "minHeight": 45,
        "padding": 10,
      }
    }
    testID="button"
  >
    <Text
      style={
        {
          "fontSize": 16,
          "fontWeight": "bold",
          "textAlign": "center",
          "width": "auto",
        }
      }
    >
      Click Me
    </Text>
  </View>
</View>
`;
